### DISCUSSION TITLE
Theme Incorrectly Highlights Non-Existent Variables

### DISCUSSION THREAD

**Discussion by phuongpd2k** [olimorris/onedarkpro.nvim#257] [Q&A] [answered]
I’ve noticed that the theme is highlighting non-existent variables as if they were valid, which can cause confusion. I haven’t tried other color schemes, so I’m unsure if this is specific to the theme in use.

Expected Behavior
Non-existent variables should not be highlighted as if they are valid.

Nvim: 
<img width="664" alt="image" src="https://github.com/user-attachments/assets/adf44f2c-5a3e-4042-bf95-eb70de645eea">

Vscode:
<img width="829" alt="image" src="https://github.com/user-attachments/assets/1117eff6-9e27-4b74-91da-ca5001158542">

**Comment by olimorris** [comment, marked as answer]
This will be because of Tree-sitter and how it highlights your filetype. You can try changing the theme's highlight groups to suit your needs, maybe semantic token highlighting could be used as well.
